src/_P108_DDS238.ino

```diff
@@ -382,38 +382,40 @@ int p108_storageValueToBaudrate(uint8_t baudrate_setting) {
 
 float p108_readValue(uint8_t query, struct EventStruct *event) {
   uint8_t errorcode     = -1; // DF - not present in P085
-  float value = 0; // DF - not present in P085
+  float value = 0.0f; // DF - not present in P085
   P108_data_struct *P108_data =
     static_cast<P108_data_struct *>(getPluginTaskData(event->TaskIndex));
 
   if ((nullptr != P108_data) && P108_data->isInitialized()) {
     switch (query) {
       case P108_QUERY_V:
-        value = P108_data->modbus.readHoldingRegister(0x0C ,errorcode) / 10.0; // 0.1 V => V
+        value = P108_data->modbus.readHoldingRegister(0x0C ,errorcode) / 10.0f; // 0.1 V => V
         break;
       case P108_QUERY_A:
-        value = P108_data->modbus.readHoldingRegister(0x0D, errorcode) / 100.0; // 0.01 A => A
+        value = P108_data->modbus.readHoldingRegister(0x0D, errorcode) / 100.0f; // 0.01 A => A
         break;
       case P108_QUERY_W:
-        value =  P108_data->modbus.readHoldingRegister(0x0E, errorcode) * 1.0 ;
+        value =  P108_data->modbus.readHoldingRegister(0x0E, errorcode) * 1.0f;
+        if (value > 32767) { value = 0 - (65535 - value); }
         break;
       case P108_QUERY_VA:
-        value = P108_data->modbus.readHoldingRegister(0x0F, errorcode) * 1.0 ;
+        value = P108_data->modbus.readHoldingRegister(0x0F, errorcode) * 1.0f;
```

Comment:
Why multiplying with 1.0f?
If you assign it to a float, it will automatically be cast to a float.

Reply:
The multiplication was already there, I just added the `f`, but it can be removed if that's better.